packages/docregistry/src/context.ts

```diff
@@ -289,22 +289,23 @@ export class Context<
   /**
    * Save the document to a different path chosen by the user.
    *
-   * It will be rejected if the user abort providing a new path.
+   * Returns true if the file was successfully saved, false if the operation was cancelled
```

Comment:
One concern would be whether extension/users authors have sufficient information to discriminate between:
- save not being performed because user cancelled a dialog 
- save failing because there was an issue on the server

I think with the current implementation the former case will return `false` and latter will return `true`, which is not great. This is because `this.save()` and `this._finishSaveAs(newPath)` handle any exceptions themselves.

I think that at a minimum we should document the behaviour of this PR better.

```suggestion
   * Returns false if the operation was cancelled by the user.
   * 
   *
   * If operation proceeds but fails due to a content manager error,
   * the return value will still be `true`, but the actual result can be monitored
   * by listening to the {@link saveState} signal. 
```

Reply:
I see what you’re saying. But in your latter suggestion, how would users know whether the save actually succeeded? If an internal error occurred during `this.save()` that didn’t get thrown, the returned object might look like:
```ts
{
  cancelled: false,
  error: undefined
}
```
which would effectively be equivalent to returning `{ success: true }`.

So I think it might be clearer to go with your initial suggestion of documenting the behavior better.

Reply:
Well the idea then would be to duplicate the internal logic changing it so that it does throw an error, which I don't like very much either.